### PR TITLE
#244 refactored announcements => changed naming to reminders

### PR DESCRIPTION
**Pull request by AlexanderGipp**
**This does not change any functionality and only changes the naming of the module.**

### CODE REVIEW

**Review by MarkusKgit — CHANGES_REQUESTED** [review]
This needs a database migration

**Comment by AlexanderGipp** [comment]
> 
> 
> This needs a database migration

Opps i totally forgot, thank you for **reminding** me 😅

**Comment by MarkusKgit** [comment]
No worries :-) But can we do a [rename](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.migrations.migrationbuilder.renametable?view=efcore-5.0) instead of dropping the table?

**Comment by AlexanderGipp** [comment]
> 
> 
> No worries :-) But can we do a [rename](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.migrations.migrationbuilder.renametable?view=efcore-5.0) instead of dropping the table?

Oh damn you are pushing my ef core knowledge to its limit. Let me look into that though 😄

**Comment by AlexanderGipp** [comment]
> 
> 
> No worries :-) But can we do a [rename](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.migrations.migrationbuilder.renametable?view=efcore-5.0) instead of dropping the table?

Does this look solid to you?

```cs
 protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.RenameTable(
                name: "Announcements",
                newName: "Reminder");
        }

        protected override void Down(MigrationBuilder migrationBuilder)
        {

            migrationBuilder.RenameTable(
                name: "Reminder",
                newName: "Announcements");
        }
```

**Comment by MarkusKgit** [comment]
To be honest I never had to manually touch the migrations before ;-) Looks good but I think since you renamed some columns to the correct casing (ID -> Id) it will need a couple of column renames as well. 
```cs
migrationBuilder.RenameColumn(
                name: "ID",
                table: "Reminder",
                newName: "Id");
```

But if it turns out to be too much hassle we can leave it as is. I have no pending reminders in my db instance. However I don't know if anyone else is actively using the Bot and might have some :-)

**Comment by AlexanderGipp** [comment]
> 
> 
> To be honest I never had to manually touch the migrations before ;-) Looks good but I think since you renamed some columns to the correct casing (ID -> Id) it will need a couple of column renames as well.
> 
> ```cs
> migrationBuilder.RenameColumn(
>                 name: "ID",
>                 table: "Reminder",
>                 newName: "Id");
> ```
> 
> But if it turns out to be too much hassle we can leave it as is. I have no pending reminders in my db instance. However I don't know if anyone else is actively using the Bot and might have some :-)

I will glady change it, however you will need to test those changes and make sure everything still works as expected.
Just let me know if i should contiue with the migration script thingy :)

**Comment by MarkusKgit** [comment]
Sure, if you push the changes I can set up my testing Bot with actual values in the DB and try it out

File: MonkeyBot/Migrations/20211005163642_UpdatedReminders.cs [hunk 1]
```diff
@@ -7,52 +7,48 @@ public partial class UpdatedReminders : Migration
...
-                constraints: table =>
-                {
-                    table.PrimaryKey("PK_Announcements", x => x.ID);
-                });
+            migrationBuilder.RenameTable(
+                name: "Reminder",
+                newName: "Announcements");
+
+            migrationBuilder.RenameColumn(
+                table: "Reminder",
```
Comment (MarkusKgit) [inline MonkeyBot/Migrations/20211005163642_UpdatedReminders.cs:39]: table name should be Announcements here or the RenameTable needs to be the last step.

File: MonkeyBot/Database/MonkeyDBContext.cs [hunk 2]
```diff
@@ -12,7 +13,7 @@ public class MonkeyDBContext : DbContext
         public DbSet<BenzenFact> BenzenFacts { get; set; }
         public DbSet<GuildConfig> GuildConfigs { get; set; }
         public DbSet<TriviaScore> TriviaScores { get; set; }
-        public DbSet<Announcement> Announcements { get; set; }
+        public DbSet<Reminder> Reminder { get; set; }
```
Comment (MarkusKgit) [inline MonkeyBot/Database/MonkeyDBContext.cs:16]: Can we please pluralize this, so it is consistent with the others? (`Reminder` -> `Reminders`)

File: MonkeyBot/Migrations/20211005163642_UpdatedReminders.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+namespace MonkeyBot.Migrations
+{
+    public partial class UpdatedReminders : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
+            migrationBuilder.RenameTable(
+                name: "Announcements",
+                newName: "Reminder");
```
Comment (MarkusKgit) [inline MonkeyBot/Migrations/20211005163642_UpdatedReminders.cs:13]: Pluralize table name Reminders

**Review by MarkusKgit — CHANGES_REQUESTED** [review]

**Review by MarkusKgit — APPROVED** [review]

**Comment by MarkusKgit** [comment]
Everything mint now, thank you very much

**Pull request merged**